src/parser.lalrpop

```diff
@@ -36,11 +42,15 @@ pub Expr: Expression = {
 
 BaseExpr: Expression = {
 	// TODO: other keywords beside virtual?
-	<v: "virtual"?> <mut d: Decl> => {
+	<start: @L> <v: "virtual"?> <mut d: Decl> <end: @R> =>? {
```

Comment:
I think the error message will look better if we just put the range around the `virtual` keyword instead of the whole decl.